lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -551,13 +589,41 @@ def _log_notices(self, cursor):
 
 
 class PGCompiler_psycopg2(PGCompiler):
-    pass
+    execute_values_insert_template = None
+
+    # Override SQLCompiler.generate_values_placeholders_str - enable use of
+    # psycopg2 execute_values()
+    def generate_values_placeholders_str(
+            self, crud_params, returning_clause_exists):
+
+        # Currently not using psycopg2.execute_values() when there's a
+        # returning clause; need to add support
+        # for receiving multiple return values from insert query
+        # Note: self.inline is true iff there are multiple parameters sets to
+        # the query
+        if self.inline and not returning_clause_exists and \
+                self.dialect.psycopg2_executemany_mode is EXECUTEMANY_VALUES:
+            self.execute_values_insert_template = "(" + \
```

Comment:
**mike bayer** (zzzeek) wrote:

this is the string we would generate anyway..right?   Why not have the base compiler just store this string in all cases if that's the case, then we don't need this method here.